Content.Server/Traits/Assorted/EmotionLoopSystem.cs

```diff
@@ -0,0 +1,46 @@
+using Robust.Shared.Random;
+using Content.Server.Chat.Systems;
+
+namespace Content.Server.Traits.Assorted;
+
+/// <summary>
+/// This system allows triggering any emotion at random intervals.
+/// </summary>
+public sealed class EmotionLoopSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<EmotionLoopComponent, ComponentStartup>(SetupTimer);
+    }
+
+    private void SetupTimer(EntityUid uid, EmotionLoopComponent component, ComponentStartup args)
+    {
+        component.NextIncidentTime = TimeSpan.FromSeconds(_random.NextFloat(component.MinTimeBetweenEmotions, component.MaxTimeBetweenEmotions));
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<EmotionLoopComponent>();
+        while (query.MoveNext(out var uid, out var emotionLoop))
```

Comment:
This runs every frame, I'll leave a maint for a performance guess but it's probably overkill to tick this often.

Reply:
From my basic understanding, you would use `[AutoGenerateComponentPause]` and `[AutoPausedField]` (plus maybe a `TimeOffsetSerializer`?).

Honestly I'm not super clear on all of this, but you should be able to find examples in the codebase, or just ask a maintainer in the Discord to help.

Reply:
I added the `[AutoPausedField]` attribute to the "NextIncidentTime" variable of type TimeSpan. I’m not entirely sure what it affects, but I believe I’ve implemented it as requested